tests/integration/cmd_dev_debug_test.go

```diff
@@ -40,47 +41,59 @@ var _ = Describe("odo dev debug command tests", func() {
-							CmdlineArgs:   opts,
-							NoRandomPorts: true,
-						})
-						Expect(err).ToNot(HaveOccurred())
+			When("running odo dev with debug flag and custom port mapping for port forwarding", helper.LabelPodmanIf(podman, func() {
+				var (
+					devSession helper.DevSession
+					ports      map[string]string
+				)
+				var (
+					LocalPort      = helper.GetRandomFreePort()
+					LocalDebugPort = helper.GetRandomFreePort()
+				)
+				const (
+					ContainerPort      = "3000"
+					ContainerDebugPort = "5858"
+				)
+
+				BeforeEach(func() {
+					opts := []string{"--debug", fmt.Sprintf("--port-forward=%s:%s", LocalPort, ContainerPort), fmt.Sprintf("--port-forward=%s:%s", LocalDebugPort, ContainerDebugPort)}
+					if podman {
+						// Debugging works with podman if we use --forward-localhost, but that will not use custom port-mapping;
+						// so we use --ignore-localhost to simply test if custom ports are still taken into account
+						opts = append(opts, "--ignore-localhost")
+					}
+					var err error
+					devSession, _, _, ports, err = helper.StartDevMode(helper.DevSessionOpts{
+						CmdlineArgs:   opts,
+						NoRandomPorts: true,
+						RunOnPodman:   podman,
 					})
+					Expect(err).ToNot(HaveOccurred())
+				})
 
-					AfterEach(func() {
-						devSession.Stop()
-						devSession.WaitEnd()
+				AfterEach(func() {
+					devSession.Stop()
+					devSession.WaitEnd()
+				})
+
+				It("should connect to relevant custom ports forwarded", func() {
+					By("connecting to the application port", func() {
+						helper.HttpWaitForWithStatus("http://"+ports[ContainerPort], "Hello from Node.js Starter Application!", 12, 5, 200)
 					})
-					It("should connect to relevant custom ports forwarded", func() {
-						By("connecting to the application port", func() {
-							helper.HttpWaitForWithStatus("http://"+ports[ContainerPort], "Hello from Node.js Starter Application!", 12, 5, 200)
-						})
-						By("expecting a ws connection when tried to connect on default debug port locally", func() {
-							// 400 response expected because the endpoint expects a websocket request and we are doing a HTTP GET
-							// We are just using this to validate if nodejs agent is listening on the other side
-							url := fmt.Sprintf("http://%s", ports[ContainerDebugPort])
-							Expect(url).To(ContainSubstring(LocalDebugPort))
+					By("expecting a ws connection when tried to connect on default debug port locally", func() {
+						// 400 response expected because the endpoint expects a websocket request and we are doing a HTTP GET
+						// We are just using this to validate if nodejs agent is listening on the other side
+						url := fmt.Sprintf("http://%s", ports[ContainerDebugPort])
+						Expect(url).To(ContainSubstring(LocalDebugPort))
+						if !podman {
+							// this check doesn't work for podman unless we use --forward-localhost,
+							// but using it beats the purpose of testing with custom port mapping, so we skip this check
```

Comment:
Not sure I understand why `--forward-localhost` beats the purpose of testing with custom port mapping..  To me, the local port forwarded to the container debug port should be reachable with custom port mapping (if we are running with `--forward-localhost`), and we need to ensure that, no?
`--forward-localhost` is just a way to make port forwarding work on Podman just like on cluster, without having to change the application to expose sensitive ports like debug ones. So it should work with custom port mapping on Podman as well, IMO.

Reply:
My bad. I'd misunderstood the way --forward-localhost worked. I'll fix it.